src/test/java/teammates/test/cases/webapi/GetCourseActionTest.java

```diff
@@ -29,38 +30,57 @@ protected String getRequestMethod() {
     @Override
     protected void testExecute() throws Exception {
         InstructorAttributes instructor1OfCourse1 = typicalBundle.instructors.get("instructor1OfCourse1");
-        CourseAttributes courseAttributes = logic.getCourse(instructor1OfCourse1.getCourseId());
+        CourseAttributes expectedCourse = logic.getCourse(instructor1OfCourse1.getCourseId());
 
         loginAsInstructor(instructor1OfCourse1.googleId);
 
         ______TS("Not enough parameters");
 
         verifyHttpParameterFailure();
 
-        ______TS("typical success case");
+        ______TS("typical success case for instructor");
 
         String[] params = {
                 Const.ParamsNames.COURSE_ID, instructor1OfCourse1.getCourseId(),
         };
-        GetCourseAction a = getAction(params);
-        JsonResult r = getJsonResult(a);
+        GetCourseAction getCourseAction = getAction(params);
+        JsonResult response = getJsonResult(getCourseAction);
+
+        assertEquals(HttpStatus.SC_OK, response.getStatusCode());
+        CourseData courseData = (CourseData) response.getOutput();
+
+        assertEquals(expectedCourse.getId(), courseData.getCourseId());
+        assertEquals(expectedCourse.getName(), courseData.getCourseName());
+        assertEquals(expectedCourse.getTimeZone().getId(), courseData.getTimeZone());
+    }
+
+    @Test
+    protected void testExecute_nonExistentCourse_shouldFail() {
+        InstructorAttributes instructor1OfCourse1 = typicalBundle.instructors.get("instructor1OfCourse1");
+        loginAsInstructor(instructor1OfCourse1.googleId);
 
-        assertEquals(HttpStatus.SC_OK, r.getStatusCode());
-        CourseInfo.CourseResponse response = (CourseInfo.CourseResponse) r.getOutput();
+        String[] params = {
+                Const.ParamsNames.COURSE_ID, "fake-course",
+        };
 
-        assertEquals(courseAttributes.getId(), response.getCourseId());
-        assertEquals(courseAttributes.getName(), response.getCourseName());
-        assertEquals(courseAttributes.getTimeZone().getId(), response.getTimeZone());
+        assertNull(logic.getCourse("fake-course"));
+        EntityNotFoundException entityNotFoundException = assertThrows(EntityNotFoundException.class,
+                () -> getAction(params).execute());
+        assertEquals("No course with id: fake-course", entityNotFoundException.getMessage());
     }
 
     @Test
     @Override
     protected void testAccessControl() throws Exception {
```

Comment:
What if a `student` or `unregistered user` calls this endpoint?

Reply:
The `verifyOnlyInstructorsOfTheSameCourseCanAccess` call (at the end of the method) checks for student and unregistered users.